src/Adapter/MSTest.CoreAdapter/Execution/TestExecutionManager.cs

```diff
@@ -185,6 +185,8 @@ internal void SendTestResults(TestCase test, UnitTestResult[] unitTestResults, D
                     // Ignore this exception
                 }
             }
+
+            test.Id = orginalTestCaseId;
```

Comment:
Please put comments above to explain why this is required.
This is being done only for the record result ?